reddit.js

```diff
@@ -2,4 +2,49 @@ var reddit = require("./data.json")
 
 // Write your solutions below
 
-console.log(reddit["data"]);
\ No newline at end of file
+var children = reddit.data.children;
+// var childTitles = children[0].data.title;
+// var childLinks = children[0].data.permalink;
+// var childText = children[0].data.selftext;
+// var points = children[0].data.score;
+
+// console.log(childTitles);
+
+
+// QUESTION 1
+children.forEach(function(titles) {
+	console.log('Title: ' + titles.data.title);
+});
+
+// QUESTION 2
+var links = children.map(function(link) {
+	return link.data.permalink;
+});
+
+links.forEach(function(url) {
+	console.log('http://reddit.com' + url);
+});
+
+// QUESTION 3
+var hasText = function(item) {
+	return item.data.selftext.length > 0;
+};
+
+var hasSelfText = children.filter(hasText);
+
+console.log(hasSelfText);
+
+// QUESTION 4
+
+scoreArray = [];
```

Comment:
Making a scoreArray wasn't a bad idea, but we can actually just call reduce on children here.
```
var sum = children.reduce(function(runningTotal, child, index) {
	return runningTotal + child.data.score
}, 0);
console.log(sum);
```